llmfoundry/models/mpt/configuration_mpt.py

```diff
@@ -222,9 +222,9 @@ def _validate_config(self) -> None:
             raise NotImplementedError(
                 'alibi only implemented with torch and triton attention.')
         if self.attn_config['attn_uses_sequence_id'] and self.attn_config[
-                'attn_impl'] not in ['torch', 'triton']:
+                'attn_impl'] not in ['torch', 'triton'] and not (self.attn_config['attn_impl']=='flash' and is_flash_v2_installed(v2_version='v2.1.2')):
```

Comment:
```suggestion
                'attn_impl']=='flash' and not is_flash_v2_installed(v2_version='v2.1.2')):
```

the current gating logic is pretty confusing.

Reply:
The reason I kept it that way was so that if in the future we add another type of attention which does not support sequence_id masking, then this will fail. If it does support, then it can simply be added to the list ['torch', 'triton']. Maybe I overthought lol.